Custom boost installation makes compilation fail

# How to reproduce the bug

## Custom boost installation

```bash
wget https://boostorg.jfrog.io/artifactory/main/release/1.71.0/source/boost_1_71_0.tar.gz
tar xf boost_1_71_0.tar.gz 
cd boost_1_71_0
./bootstrap.sh --prefix=/builds/pubs/boost --with-libraries=program_options,filesystem,test
./b2
./b2 install
```

## Custom GUDHI installation

```bash
wget https://github.com/GUDHI/gudhi-devel/releases/download/tags%2Fgudhi-release-3.4.1.post2/gudhi.3.4.1.post2.tar.gz
tar xf gudhi.3.4.1.post2.tar.gz 
cd gudhi.3.4.1.post2/
```

According to [CGAL's Configuration Variables](https://doc.cgal.org/latest/Manual/configurationvariables.html#title8)

```bash
cmake -DCMAKE_BUILD_TYPE=Release -DBoost_DIR=/builds/pubs/boost/lib/cmake/Boost-1.71.0 ..
make VERBOSE=1
```

# Observed behaviour

```
STEM_NO_DEPRECATED -DCGAL_EIGEN3_ENABLED -I/usr/include/x86_64-linux-gnu -I/builds/install/gudhi.3.4.1.post2/include -isystem /builds/pubs/eigen/include/eigen  -Wall -O3 -DNDEBUG   -std=gnu++14 -o CMakeFiles/periodic_cubical_complex_persistence.dir/periodic_cubical_complex_persistence.cpp.o -c /builds/install/gudhi.3.4.1.post2/utilities/Bitmap_cubical_complex/periodic_cubical_complex_persistence.cpp
In file included from /builds/install/gudhi.3.4.1.post2/include/gudhi/reader_utils.h:14:0,
                 from /builds/install/gudhi.3.4.1.post2/utilities/Bitmap_cubical_complex/periodic_cubical_complex_persistence.cpp:11:
/builds/install/gudhi.3.4.1.post2/include/gudhi/graph_simplicial_complex.h:14:10: fatal error: boost/graph/adjacency_list.hpp: No such file or directory
 #include <boost/graph/adjacency_list.hpp>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

# Expected behaviour

It should compile

# Fix proposal

in `cmake/modules/GUDHI_third_party_libraries.cmake`:
```
...

# Boost_FOUND is not reliable
if(NOT Boost_VERSION)
  message(FATAL_ERROR "NOTICE: This program requires Boost and will not be compiled.")
endif(NOT Boost_VERSION)
### BUG FIX
include_directories(${Boost_INCLUDE_DIRS})
### END OF BUG FIX
```

--- a/src/cmake/modules/GUDHI_third_party_libraries.cmake
+++ b/src/cmake/modules/GUDHI_third_party_libraries.cmake
@@ -6,6 +6,7 @@ find_package(Boost 1.56.0 QUIET OPTIONAL_COMPONENTS filesystem unit_test_framewo
 if(NOT Boost_VERSION)
   message(FATAL_ERROR "NOTICE: This program requires Boost and will not be compiled.")
 endif(NOT Boost_VERSION)
+include_directories(${Boost_INCLUDE_DIRS})
 
 find_package(GMP)
 if(GMP_FOUND)

